tls/s2n_cipher_suites.c

```diff
@@ -1155,17 +1173,183 @@ int s2n_set_cipher_as_client(struct s2n_connection *conn, uint8_t wire[S2N_TLS_C
     return 0;
 }
 
-static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, uint32_t count, uint32_t cipher_suite_len)
+/* This function returns the client index for a cipher *match. If a matching cipher could not be found, then -1 is returned. */
+static int s2n_wire_ciphers_has_server_cipher_at(const uint8_t *match, const uint8_t *wire, uint32_t count, uint32_t cipher_suite_len)
 {
+    int negative_one = -1;
     for (uint32_t i = 0; i < count; i++) {
         const uint8_t *theirs = wire + (i * cipher_suite_len) + (cipher_suite_len - S2N_TLS_CIPHER_SUITE_LEN);
 
-        if (!memcmp(match, theirs, S2N_TLS_CIPHER_SUITE_LEN)) {
-            return 1;
+        if (s2n_constant_time_equals(match, theirs, S2N_TLS_CIPHER_SUITE_LEN)) {
+            return i;
         }
     }
 
-    return 0;
+    return negative_one;
+}
+
+static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, uint32_t count, uint32_t cipher_suite_len)
```

Comment:
Trying to rename this variable since a match hasn't been found yet, if I understand correctly. It's just a potentially match. Also this function should return a boolean.
```suggestion
static bool s2n_wire_ciphers_contain(const uint8_t *iana_value, const uint8_t *wire, uint32_t count, uint32_t cipher_suite_len)
```
Also can you add a tab to the wrapped line?

Reply:
This is a bit out-of-scope. This function already existed, I just modified the implementation because it was recommended by the comments to remove duplicating existing code. I'm going to leave the return value and parameter names as is.

Reply:
I disagree. Anytime you touch old code you should be improving it, and in this case, it doesn't make any sense for this function to return an int. As long as it's not a huge change you should fix it.